src/HttpHeader.cc

```diff
@@ -747,6 +748,20 @@ HttpHeader::refreshMask()
     }
 }
 
+size_t
+HttpHeader::maxLen() const
+{
+    switch (owner)
+    {
+        case hoRequest:
+            return Config.maxRequestHeaderSize;
+        case hoReply:
+            return Config.maxReplyHeaderSize;
+        default:
+            return 0; // not restricted
```

Comment:
We do not document support for special zero reply_header_max_size or request_header_max_size values, and we probably do not support them in some existing code. I think we should not start.

Please use std::numeric_limits<size_t>::max() instead of a special value (and simplify the method documentation/caller accordingly).